apps/src/templates/manageStudents/ManageStudentsLoginInfo.jsx

```diff
@@ -0,0 +1,75 @@
+import React, {Component, PropTypes} from 'react';
+import LoginTypeParagraph from '@cdo/apps/templates/teacherDashboard/LoginTypeParagraph';
+import {SectionLoginType} from '@cdo/apps/util/sharedConstants';
+import i18n from "@cdo/locale";
+
+const styles = {
+  explination: {
+    clear: 'both',
+    paddingTop: 20,
+  }
+};
+
+class ManageStudentsLoginInfo extends Component {
+  static propTypes = {
+    sectionId: PropTypes.number,
+    sectionCode: PropTypes.string,
+    loginType: PropTypes.string,
+  };
+
+  render() {
+    const {loginType, sectionId, sectionCode} = this.props;
+    return (
+      <div style={styles.explination}>
+        <h2>Login information</h2>
+        {(loginType === SectionLoginType.word || loginType === SectionLoginType.picture) &&
+          <div>
+            <p>
+              {i18n.sectionCode() + ': ' + sectionCode}
+            </p>
+            <p>
+              {i18n.joinSectionExplination()}
+              <a target="_blank" href={`${window.location.origin}/join`}>
+                {`${window.location.origin}/join`}
+              </a>
+            </p>
+            <p>
+              {i18n.sectionSignInInfo()}
+              <a target="_blank" href={`http://localhost-studio.code.org:3000/sections/${sectionCode}`}>
+                {`http://localhost-studio.code.org:3000/sections/${sectionCode}`}
+              </a>
+            </p>
+            <p>
+              <a target="_blank" href={`/teacher-dashboard#/sections/${sectionId}/print_signin_cards`}>
+                {i18n.printLoginCardExplination()}
+              </a>
+            </p>
+          </div>
+        }
+        {(loginType === SectionLoginType.email) &&
+          <div>
+            <p>
+              {i18n.joinSectionAsk()}
+              <a target="_blank" href={`${window.location.origin}/join/${sectionCode}`}>
+                {`http://localhost-studio.code.org:3000/join/${sectionCode}`}
```

Comment:
I'm struggling to figure out how to make these URLs dynamic based on environment, since there are both studio and code.org urls and this is from apps in pegasus. Any ideas are welcome.

Reply:
Generally speaking, you'll want to find the right method in deployment.rb (possibly `dashboard_hostname`), pass it into your js entry point in the data tag on the script element, and then pass it through props into your react component. I'll try to dig up an example.

Reply:
looks like `studioUrlPrefix` is already available here: https://github.com/code-dot-org/code-dot-org/blob/f372627ef22dc536b65e8c8ae93b627dec0ce8c7/apps/src/sites/code.org/pages/public/teacher-dashboard/index.js#L39